### PR TITLE
DOCS-1678-Add LDAP example

### PR DESCRIPTION
**Pull request by bmckercher123**
Add LDAP troubleshooting to IDP doc

Product Version(s):
CE 3.17, 3.18

Issue:
https://tigera.atlassian.net/browse/DOCS-1678

Link to docs preview:

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [x] Deploy preview inspected wherever changes were made 
- [x] Build completed successfully
- [x] Test have passed

### CODE REVIEW

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx [hunk 1]
```diff
@@ -31,6 +31,7 @@ When configuring your cluster, you may be asked for the following inputs:
 - **OIDC authentication**: User identity is managed outside of the cluster by an OIDC authorization server.
 - **Google OIDC authentication**: User identity is managed by Google OIDC. Choose this option if you want to use GSuite groups.
 - **OpenShift authentication**: User identity is provided by the OpenShift OAuth server.
+- **LDAP authentication**: User identify is provided using Active Directory.
```
Comment (rene-dekker) [inline calico-enterprise/operations/cnx/configure-identity-provider.mdx]: identity*

I would say ldap server, not active directory, even though AD is the most common application of ldap.

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx [hunk 2]
```diff
@@ -42,215 +43,242 @@ When configuring your cluster, you may be asked for the following inputs:
...
-   `CLUSTER_DOMAIN` is the domain (excl. port) where your OpenShift cluster is accessed (run `oc status` to get it) and `CLIENT_SECRET` is a value of your choosing.
-
-   ```bash
+  <TabItem label="OIDC" value="OIDC-0">
+
+    1. Apply the Authentication CR to your cluster to let the operator configure your login.
+    The following example uses the email claim. The email field from the JWT (created by your IdP), is used as the username for binding privileges.
+
+    ```yaml
+    apiVersion: operator.tigera.io/v1
```
Comment (rene-dekker) [inline calico-enterprise/operations/cnx/configure-identity-provider.mdx]: This yaml and all others have lost their tabs.

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx [hunk 3]
```diff
@@ -42,214 +43,242 @@ When configuring your cluster, you may be asked for the following inputs:
...
+    rootCA: <your-base64-ca-cert-pem>
+    ```
+
+    ### Troubleshooting LDAP
+
+    **Problem**: Error on Manager UI: 502 bad gateway nginx
+
+    **Solution/workaround**: If you have many groups of users, you may need to add filters in the AD configuration to reduce the number of groups that are passed to {{prodname}}. The issue is that nginx ingress cannot handle the length of the token. To solve this issue, apply a filter: `(&(objectClass=group)(|(cn:dn:=HwInf-K8S-Support)(cn:dn:=HWInf-CISD)))` as shown in the following example.
+
+    ```yaml
```
Comment (rene-dekker) [inline calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: Like all other yamls, this is malformatted.

I think that the mentions of "HwInf-K8S-Support" and "HWInf-CISD"  should be censored. For example "example-dn1" and "example-dn2"

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx [hunk 4]
```diff
@@ -42,214 +43,242 @@ When configuring your cluster, you may be asked for the following inputs:
...
+    nameAttribute: cn
+    userMatchers:
+    - groupAttribute: member
+    userAttribute: distinguishedName
+    host: ldap.gtm.mycompany.com:636
+    startTLS: false
+    userSearch:
+    baseDN: OU=Accounts,DC=mycompany,DC=com
+    nameAttribute: userPrincipalName
+    managerDomain: https://tigera-manager.sc02-paas.mycompany.com
```
Comment (rene-dekker) [inline calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: I would remove `sc02-paas`

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx [hunk 5]
```diff
@@ -42,214 +43,242 @@ When configuring your cluster, you may be asked for the following inputs:
...
+    spec:
+    ldap:
+    groupSearch:
+    baseDN: OU=GroupID,OU=Groups,DC=mycompany,DC=com
+    filter: (&(objectClass=group)(|(cn:dn:=HwInf-K8S-Support)(cn:dn:=HWInf-CISD)))
+    nameAttribute: cn
+    userMatchers:
+    - groupAttribute: member
+    userAttribute: distinguishedName
+    host: ldap.gtm.mycompany.com:636
```
Comment (rene-dekker) [inline calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: I would remove `.gtm`

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx (hunk 1 above)
Comment (bmckercher123) [reply calico-enterprise/operations/cnx/configure-identity-provider.mdx]: Fixed.

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx (hunk 3 above)
Comment (bmckercher123) [reply calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: Fixed.

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx (hunk 4 above)
Comment (bmckercher123) [reply calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: Removed.

File: calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx (hunk 5 above)
Comment (bmckercher123) [reply calico-enterprise_versioned_docs/version-3.17/operations/cnx/configure-identity-provider.mdx]: Removed.

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx (hunk 2 above)
Comment (bmckercher123) [reply calico-enterprise/operations/cnx/configure-identity-provider.mdx]: Fixed

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx [hunk 6]
```diff
@@ -248,7 +251,33 @@ When configuring your cluster, you may be asked for the following inputs:
      bindPW: <your-base64-bind-password>
      rootCA: <your-base64-ca-cert-pem>
    ```
-###
+  ### Troubleshooting LDAP
+
+  **Problem**: Error: (502 bad gateway nginx) when logging in to Manager UI
+
+  **Solution**: nginx ingress cannot handle the length of the token when adding a large number of groups to {{prodname}}.
+  To resolve this, apply a `filter:` as shown in the following example.
```
Comment (rene-dekker) [inline calico-enterprise/operations/cnx/configure-identity-provider.mdx]: How about just saying "apply a filter" or "apply  a filter section" ?

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx [hunk 7]
```diff
@@ -248,7 +251,33 @@ When configuring your cluster, you may be asked for the following inputs:
      bindPW: <your-base64-bind-password>
      rootCA: <your-base64-ca-cert-pem>
    ```
-###
+  ### Troubleshooting LDAP
+
+  **Problem**: Error: (502 bad gateway nginx) when logging in to Manager UI
+
+  **Solution**: nginx ingress cannot handle the length of the token when adding a large number of groups to {{prodname}}.
```
Comment (rene-dekker) [inline calico-enterprise/operations/cnx/configure-identity-provider.mdx]: > when adding a large number of groups to {{prodname}}.

How about "when a `groupSearch` matches too many groups."

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx (hunk 6 above)
Comment (bmckercher123) [reply calico-enterprise/operations/cnx/configure-identity-provider.mdx]: Fixed.

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx (hunk 7 above)
Comment (bmckercher123) [reply calico-enterprise/operations/cnx/configure-identity-provider.mdx]: Good one. Fixed.

File: calico-enterprise/operations/cnx/configure-identity-provider.mdx [hunk 8]
```diff
@@ -248,7 +251,33 @@ When configuring your cluster, you may be asked for the following inputs:
...
+
+  ```yaml
+  apiVersion: operator.tigera.io/v1
+  kind: Authentication
+    name: tigera-secure
+  spec:
+    ldap:
+      groupSearch:
+        baseDN: OU=GroupID,OU=Groups,DC=mycompany,DC=com
+        filter: (&(objectClass=group)(|(cn:example-dn1)(cn:example-dn2)))
```
Comment (rene-dekker) [inline calico-enterprise/operations/cnx/configure-identity-provider.mdx]: I see I made a mistake when I was behind your desk:
```
filter: (&(objectClass=group)(|(cn:example-dn1)(cn:example-dn2)))
```

Shoud be

```
filter: (&(objectClass=group)(|(cn=example-dn1)(cn=example-dn2)))
```

**Review by rene-dekker — APPROVED** [review]
lgtm after the comment is addressed

**Pull request merged**